EEJT-0003/src/main/java/com/epam/esm/giftcertficate/repository/GiftCertificateRepository.java

```diff
@@ -0,0 +1,26 @@
+package com.epam.esm.giftcertficate.repository;
+
+import com.epam.esm.giftcertficate.model.GiftCertificate;
+import com.epam.esm.utils.query.SqlQuery;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface GiftCertificateRepository extends JpaRepository<GiftCertificate, Long> {
+
+    @Query(SqlQuery.GiftCertificate.IS_GIFT_CERTIFICATE_EXISTS_BY_NAME)
+    boolean isGiftCertificateExistByName(@Param("name") String name);
```

Comment:
```suggestion
    boolean existsByName(String name);
```
The JpaRepository has an internal system that analyses the method name you have written and then generates the query that has to be executed to the database.
Its can work without @Query annotations check it.

Reply:
wow , it is really good , ty